readthedocs/rtd_tests/tests/test_backend.py

```diff
@@ -78,6 +78,15 @@ def test_parse_git_tags(self):
                      self.project.vcs_repo().parse_tags(data)]
         self.assertEqual(expected_tags, given_ids)
 
+    def test_check_for_submodules(self):
+        repo = self.project.vcs_repo()
+
+        repo.checkout()
+        self.assertFalse(repo.submodules_exists())
+
+        repo.checkout('submodule')
```

Comment:
Can you add a comment here explaining why you checkout submodule here and what's the difference with the other branch? Just for future reference.